### ISSUE TITLE
products.deleteMany()` buffering timed out after 10000ms

### ISSUE THREAD

**Issue by nyashamajome** [Automattic/mongoose#9846]
Hi i am trying to upload my products from my json file to the database. I am able to connect to the database however when i try to upload the products i am getting this error. 

```
Error with data import MongooseError: Operation `products.deleteMany()` buffering timed out after 10000ms
    at Timeout.<anonymous> (C:\Users\nmajo\OneDrive\Desktop\Webapp2\node_modules\mongoose\lib\drivers\node-mongodb-native\collection.js:185:20)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! Webapp2@1.0.0 data:import: `node backend/DBScript.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the Webapp2@1.0.0 data:import script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\nmajo\AppData\Roaming\npm-cache\_logs\2021-01-21T23_53_15_306Z-debug.log
```

This is the code i am using. 

```javascript
require("dotenv").config();
const productData = require("./data/products");
const connectDB = require("./config/database");
const Product = require("./models/Products");

 connectDB();

const importData = async () => {
  try {
    await Product.deleteMany({});

    await Product.insertMany(productData);

    console.log("Data Import Success");

    process.exit();
  } catch (error) {
    console.error("Error with data import", error);
    process.exit(1);
  }
};
importData();
```

When i remove the await on the Product.deleteMany and the Insert many, its saying products have been uploaded, however when i check the mongo db, there are no products. Does anyone have an idea on how i can fix this ?

**Comment by vkarpov15** [comment]
Can you please paste what `config/database.js` and `models/Products.js` contain?

**Comment by nashamaj** [comment]
So the code for the config/database is 

require("dotenv").config();
const mongoose = require("mongoose");

const connectDB = async () => {
  try {
    await mongoose.connect(process.env.MONGO_URI, {
      useNewUrlParser: true,
      useUnifiedTopology: true,
      useCreateIndex: true,
      useFindAndModify: false,
    });

    console.log("MongoDB connection SUCCESS");
  } catch (error) {
    console.error("MongoDB connection FAIL");
    process.exit(1);
  }
};

module.exports = connectDB;

and for the models/products.js  is 

const mongoose = require("mongoose");

const productSchema = mongoose.Schema({
  name: {
    type: String,
    required: true,
  },
  description: {
    type: String,
    required: true,
  },
  price: {
    type: Number,
    required: true,
  },
  countInStock: {
    type: Number,
    required: true,
  },
  imageUrl: {
    type: String,
    required: true,
  },
});

const Product = mongoose.model("product", productSchema);

module.exports = Product;

**Comment by nyashamajome** [comment]
> Can you please paste what `config/database.js` and `models/Products.js` contain?

Hi @vkarpov15 

So the code for the config/database.js is 



``
  
require("dotenv").config();
const mongoose = require("mongoose");

const connectDB = async () => {
  try {
    await mongoose.connect(process.env.MONGO_URI, {
      useNewUrlParser: true,
      useUnifiedTopology: true,
      useCreateIndex: true,
      useFindAndModify: false,
    });

    console.log("MongoDB connection SUCCESS");
  } catch (error) {
    console.error("MongoDB connection FAIL");
    process.exit(1);
  }
};

module.exports = connectDB;``

and the code for the products.js is

``
const mongoose = require("mongoose");

const productSchema = mongoose.Schema({
  name: {
    type: String,
    required: true,
  },
  description: {
    type: String,
    required: true,
  },
  price: {
    type: Number,
    required: true,
  },
  countInStock: {
    type: Number,
    required: true,
  },
  imageUrl: {
    type: String,
    required: true,
  },
});

const Product = mongoose.model("product", productSchema);

module.exports = Product;

**Comment by vkarpov15** [comment]
Try moving `connectDB();` into your async function:

```javascript
const importData = async () => {
  try {
    await connectDB();

    await Product.deleteMany({});

    await Product.insertMany(productData);

    console.log("Data Import Success");

    process.exit();
  } catch (error) {
    console.error("Error with data import", error);
    process.exit(1);
  }
};
importData();
```

I suspect your MongoDB connection string is incorrect, and this change will help confirm that.

**Comment by TerraAtgithub** [comment]
Hi I also have a similar issue and nothing has worked for me so far (users.findOne in my case). One thing that I noticed is that when I search "buffering timed out after 10000ms" in stackoverflow pretty much all of the problems with the above phrasing came from December 2020 and January 2021 (also Feb 2021). So perhaps this is a mongoose problem caused by an update within that time period?

**Comment by leungk712** [comment]
Seconded w/ @TerraAtgithub . I tried to deploy to Heroku many times yesterday but I kept encountering a buffering .find() 10000ms. Even when I've wrapped my mongoose.connect inside an async/await function I received a "Error: MongooseError: Cannot call posts.find() before initial connection is complete if bufferCommands = false. Make sure you await mongoose.connect() if you have bufferCommands = false."

Have noticed few posts recently about the mongoose.connect buffering 10000ms issue within the past few months.
